.circleci/config.yml

```diff
@@ -363,29 +360,38 @@ workflows:
             - build_dapp
 
   publish_production:
+    when:
+      and:
+        - not: << pipeline.git.branch >>
```

Comment:
Branch is not set for tags? Or when would we require both conditions?

Reply:
Yes. So these both variables are exclusive. The pipeline gets either triggered by a tag or a branch update/creation (there are more triggers). So either of both is empty. So we could get away with just one of both expressions here. This is just more explicit and since the recent bad experience with tags and CircleCI I wanted to be sure this is as secure as it can be. :joy: 
Note that we can't use regex for the tag here. But since each job needs a tag filter anyways to run for a tag triggered pipeline (you remember?) this would be useless anyways. So this `when` just states that this job should run when the pipeline got triggered by a tag. The tag filtering is then done at the job list of the workflow.